go/vt/vtgate/planbuilder/testdata/filter_cases.json

```diff
@@ -4820,5 +4820,100 @@
       ]
     }
+  },
+  {
+    "comment": "Between clause on primary indexed id column (binary vindex on id)",
+    "query": "select id from seq_tbl where id between 1 and 5",
+    "plan": {
+      "QueryType": "SELECT",
+      "Original": "select id from seq_tbl where id between 1 and 5",
+      "Instructions": {
+        "OperatorType": "Route",
+        "Variant": "Between",
+        "Keyspace": {
+          "Name": "sequential",
+          "Sharded": true
+        },
+        "FieldQuery": "select id from seq_tbl where 1 != 1",
+        "Query": "select id from seq_tbl where id between 1 and 5",
+        "Table": "seq_tbl",
+        "Values": [
+          "(1, 5)"
+        ],
+        "Vindex": "binary"
+      },
+      "TablesUsed": [
+        "sequential.seq_tbl"
+      ]
+    }
+  },
+{
+    "comment": "Between clause on group_id column (xxhash vindex on group_id)",
+    "query": "select id, group_id from seq_tbl where group_id between 1 and 5",
+    "plan": {
+      "QueryType": "SELECT",
+      "Original": "select id, group_id from seq_tbl where group_id between 1 and 5",
+      "Instructions": {
+        "OperatorType": "Route",
+        "Variant": "Scatter",
+        "Keyspace": {
+          "Name": "sequential",
+          "Sharded": true
+        },
+        "FieldQuery": "select id, group_id from seq_tbl where 1 != 1",
+        "Query": "select id, group_id from seq_tbl where group_id between 1 and 5",
+        "Table": "seq_tbl"
+      },
+      "TablesUsed": [
+        "sequential.seq_tbl"
+      ]
+    }
+  },
+{
+    "comment": "Between clause on col2 column (there is no vindex on this column)",
+    "query": "select id, col2 from seq_tbl where col2 between 10 and 50",
+    "plan": {
+      "QueryType": "SELECT",
+      "Original": "select id, col2 from seq_tbl where col2 between 10 and 50",
+      "Instructions": {
+        "OperatorType": "Route",
+        "Variant": "Scatter",
+        "Keyspace": {
+          "Name": "sequential",
+          "Sharded": true
+        },
+        "FieldQuery": "select id, col2 from seq_tbl where 1 != 1",
+        "Query": "select id, col2 from seq_tbl where col2 between 10 and 50",
+        "Table": "seq_tbl"
+      },
+      "TablesUsed": [
+        "sequential.seq_tbl"
+      ]
+    }
+},
+{
+  "comment": "Between clause on multicolumn vindex (cola,colb)",
+  "query": "select cola,colb,col1 from seq_multicol_tbl where cola between 1 and 5",
+  "plan": {
+    "QueryType": "SELECT",
+    "Original": "select cola,colb,col1 from seq_multicol_tbl where cola between 1 and 5",
+    "Instructions": {
+      "OperatorType": "Route",
+      "Variant": "Scatter",
+      "Keyspace": {
+        "Name": "sequential",
+        "Sharded": true
+      },
+      "FieldQuery": "select cola, colb, col1 from seq_multicol_tbl where 1 != 1",
+      "Query": "select cola, colb, col1 from seq_multicol_tbl where cola between 1 and 5",
+      "Table": "seq_multicol_tbl",
+      "Values": [
+        "(1, 5)"
```

Comment:
I don't understand why we get the values field here for a `Scatter` route. it doesn't look right

Reply:
thanks for catching this , noticed `processSingleColumnVindex` has few extra early checks compared to `processMultiColumnVindex`.  With 8a0ded00679a5e0c640dfdba0340c241a7364dc7 , added checks for vindex and also routingCode before processing any futher.

This ensures `Values` is not set in case of `Scatter` route. 

Please help to review if addition of these checks has any other side effects (which i may have missed) , ensured all test cases are passing.